swarm_conn.go

```diff
@@ -75,6 +77,26 @@ func (c *Conn) doClose() {
 		c.swarm.notifyAll(func(f network.Notifiee) {
 			f.Disconnected(c.swarm, c)
 		})
+
+		// ---------------------------------------------------------
+		// Emit the Connectedness=NotConnected event on the bus if we just went
+		// from having a/many connections with the peer to having no connection to the peer
+		p := c.RemotePeer()
+		indexForLk := len(p) - 1
+		stripe := &c.swarm.stripedPeerConnectivity[p[indexForLk]]
+		stripe.Lock()
+		c.swarm.conns.RLock()
+		// we went from having a connection/s with the peer to having no connections to it
+		// if after this disconnection, we have no OPEN connections with the peer
+		isConnectednessFlip := len(c.swarm.conns.m[p]) == 0
```

Comment:
We have to look at this _where_ we remove the connection from the map (`removeConn`). If we do this _here_, multiple connections could hit this same point at the same time, all firing the event.